main/src/library/File.flix

```diff
@@ -151,4 +198,83 @@ namespace File {
         else
             readAllHelper(r, line :: acc)
 
+    ///
+    /// Returns a list of all the lines in the given file `f` with the options `opts`.
+    /// TODO: specify the opts.
+    ///
+    pub def readLinesWith(opts: {charSet :: String}, f: String): Result[List[String], String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path, ##java.nio.charset.Charset);
+            import java.nio.charset.Charset:forName(String);
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath, forName(opts.charSet));
+            Ok(readAll(reader))
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns an iterator of the given file `f`
+    ///
+    pub def readLinesIter(f: String): Iterator[Result[String, String]] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path);
+            import java.io.BufferedReader.readLine();
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath);
+            let line = ref readLine(reader);
+
+            let done = () -> Object.isNull(deref line);
+            let next = () -> {
+                let l = deref line;
+                line := readLine(reader);
+                Ok(l)
+            };
+            Iterator(done, next)
+        } catch {
+            case ex: ##java.io.IOException =>
```

Comment:
Something like this?

Reply:
Yes, like that. Do you think we should also stop iteration after an error?

Reply:
How? What do you mean?

Reply:
It seems to me that we would return an infinite iterable of Errs if there's something wrong with the file. I'm thinking it would be better to set `line` to `null` after an error to trigger an end of iteration.

Reply:
OK, let us do it. Set line to `null` after an error.